### DISCUSSION TITLE
.Net 8 and Blazor-Webassembly

### DISCUSSION THREAD

**Discussion by domenicocanavero** [Megabit/Blazorise#5643] [Q&A]
I need to develop an app with .Net8-blazor webassembly-Blazorise, what you suggest to create the project solution:
1. VS2022 template: Blazor Web App and then add Blazorise-Bootstrap-...
Or
2. VS2022 but with the Blazorise App (Megabit) template.
?

Thanks

**Comment by stsrki** [comment]
It depends. Blazorise App will start you sooner, but with  Blazor Web App you will have to do more things manually to include Blazorise. The process is fairly easy either way. Start with the Blazorise App.

**Comment by domenicocanavero** [comment]
Thanks for quickly answer.

For personal knowledge only, can you suggest to me a tutorial to initialize a Blazor Web App with WebAssembly render mode project solution to use Blazorise and Bootstrap?

Thanks

**Reply by stsrki** [reply]
I believe you can follow this one https://www.sitepoint.com/net-8-blazor-render-modes-explained/

Just replace the server with webassemly.

**Comment by domenicocanavero** [comment]
Propably my question was not clear,
I repeat it with other words ...
**How to add Blazorise and Bootstrap5** to a solution Blazor Web App for Webassembly using the VS2022 default template. It is a solution because VS2022 creates 2 projects: host side and client side. What I need to add on both projects?

A tutorial like this:
https://blazorise.com/docs/start,
more or less, but specific for webassembly or pwa solution with Blazor and Asp.Net8.

Thanks.

**Reply by David-Moreira** [reply]
I believe our NET 8 Web app template already sets up things for you. It's just a matter of choosing the RenderMode to be Webassembly, and to move things (if there are any that you want to load in webassembly exclusively) to the client project.

**Reply by David-Moreira** [reply]
I like the new NET8 web app alot, but I don't think they managed to get the complexity down correctly for people, there's too many things that can go wrong in my opinion and make us scratch our heads.

I generally just recommend people to stick to one of the flavors, Blazor.Server and Blazor.WebAssembly, so they can actually get started without too much cerimony. (You can always migrate in the future once you have a better understanding, we've lost some time, but we managed to migrate our docs app fairly well)

Unless you **explicitly** want to make use of **SSR**, static rendering, enhanced navigation, etc... then in that case you should start with the webapp.

**Comment by domenicocanavero** [comment]
Thanks
